Add tests for ID-based `docker plugin enable/disable/rm/set`

yongtang opened the pull request:
**- What I did**

This fix is a follow up based on comment:
https://github.com/docker/docker/pull/28789#issuecomment-264225277

As #28789 has been merged in, it is possible for `docker plugin inspect` to search based on Name or ID Prefix. However, ID-based `docker plugin enable/disable/rm/set` are still not possible.

**- How I did it**

This fix allows `docker plugin enable/disable/rm/set` to search based on:
- Full ID
- Full Name
- Partial ID (prefix)

**- How to verify it**

An additional integration test has been added to cover the changes. And all existing tests should pass.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

![collection-of-christmas-kitten-wallpaper-on-wall-papers-1](https://cloud.githubusercontent.com/assets/6932348/20985763/b2c7d658-bc7a-11e6-8327-0288096d6e38.jpg)


This fix is a follow up of #28789.

/cc @anusha-ragunathan

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

thaJeztah commented:
/cc @tiborvass @vieux

tonistiigi commented:
@yongtang This should be included in #29487 . Sorry for hijacking but it has some more critical fixes that are tied into ID requests and couldn't wait. Can you double check that that PR fixes this use case and if everything is ok, rebase your PR to only add the test.

yongtang committed:
Tests to allow ID-based `docker plugin enable/disable/rm/set`
This fix is a follow up based on comment:

and a follow up to:
https://github.com/docker/docker/pull/29222#issuecomment-268908937

As #28789 has been merged in, it is possible for `docker plugin inspect`
to search based on Name or ID Prefix. However, ID-based
`docker plugin enable/disable/rm/set` are still not possible.

This fix addes test for  `docker plugin enable/disable/rm/set` to search based on:
- Full ID
- Full Name
- Partial ID (prefix)

The actual fix is done in  #29487.

This fix is a follow up of #28789 and  #29487.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
@tonistiigi Thanks. I checked the PR #29487 and verified that it addressed the issue. The following is the test pass on top of #29487:
```
INFO: Testing against a local daemon
PASS: docker_cli_plugins_test.go:269: DockerSuite.TestPluginIDPrefix	4.554s
OK: 1 passed
PASS
```

I also rebased this PR so that now it only includes tests. The test for this PR will fail temporarily but should pass once PR #29487  is merged.

tonistiigi commented:
@yongtang Needs rebase

yongtang commented:
Thanks @tonistiigi The PR has been rebased.

tonistiigi commented:
LGTM

thaJeztah reviewed (approved):
LGTM, thanks!
